matminer/featurizers/structure.py

```diff
@@ -3612,3 +3613,189 @@ def citations(self):
+            valence sum method for disordered structures
+    '''
+    
+    def __init__(self, r_cut=4.0, disordered_pymatgen=False):
+        
+        basedir = os.path.dirname(os.path.realpath(__file__))
+        bv_path = os.path.join(basedir, "../utils/data_files/Bond_valences2016.csv")
+        self.bv_values = pd.read_csv(bv_path)
+        
+        self.r_cut = r_cut
+        self.disordered_pymatgen = disordered_pymatgen
+        
+    
+    def precheck(self, struct):
+        """Bond valence methods require atom pairs with oxidation states.
+
+        Args:
+            struct: Pymatgen Structure
+        """
+        anions = ["O", "N", "F", "Cl", "Br", "S", "Se", "I", "Te", "P", "H", "As"]
+        # if structure lacks oxidation state information, fail precheck
+        if any(isinstance(site.species.elements[0], Element) for site in struct):
+            return False
+        elems = [str(x.element) for x in struct.composition.elements]
+
+        # If compound is not ionically bonded, it is going to fail
+        if not any([e in anions for e in elems]):
+            return False
+        valences = [site.species.elements[0].oxi_state for site in struct]
+
+        # If the oxidation states are technically provided but 0, return false
+        if min(valences) == 0:
+            return False
+
+        if len(struct) > 200:
+            raise UserWarning("Computing bond valence sums for over 200 sites. "
+                              "Might be slow")
+        return True
+    
+    def featurize(self, struct):
+        """
+        Get global instability index.
+        
+        Args:
+            struct: Pymatgen Structure object
+        Returns:
+            [gii]: Length 1 list with float value
+        """
+        if not struct.is_ordered:
+            if self.disordered_pymatgen:
+                gii = self.calc_gii_pymatgen(struct)
+                if gii > 0.6:
+                    raise ValueError(
+                        "GII extremely large. Pymatgen method may not be suitable "
+                        "or structure may be unusual.")
+                return [gii]
+            else:
+                raise ValueError('Structure must be ordered for table lookup method.')
+        
+        gii = self.calc_gii_iucr(struct)
+        if gii > 0.6:
+            raise Exception("GII extremely large. Table parameters may "
+                            "not be suitable or structure may be unusual.")
+        return [gii]
+        
+    def calc_gii_iucr(self, struct):
+        elements = [str(i) for i in struct.composition.element_composition.elements]
+        bond_valence_sums = []
+        cutoff = self.r_cut
+        
+        # for loop to calculate the BV sum on each site
+        for site in struct:
+            site_val = site.species.elements[0].oxi_state
+            site_el = str(site.species.element_composition.elements[0])
+            bvs = 0
+            neighbors = struct.get_neighbors(site, r=cutoff)
```

Comment:
There is a `get_all_neighbors` function for structure that could be a little faster. That said, I'm not sure how much faster and whether performance is currently a concern for your featurizer. So, regard this comment as optional.

Reply:
Thanks for letting me know. I'm going to leave as is for now to avoid refactoring the whole thing, but if performance is noticeably bad, I'll switch.